components/textfield/index.css

```diff
@@ -10,116 +10,445 @@ OF ANY KIND, either express or implied. See the License for the specific languag
   }
+}
 
-  /* focus-ring */
-  &:after {
-    pointer-events: none;
-    content: '';
+/********* Child Element - Validation Icons - ⚠️ ✅ *********/
+.spectrum-Textfield-validationIcon {
+  position: absolute;
+  pointer-events: all;
+  top: 0;
+  right: 0;
+
+  [dir="rtl"] & {
+    left: 0;
+    right: auto;
+  }
+
+  /****** Validation Icon - Valid ✅ ******/
+  .is-valid & {
+    margin-block-start: var(--mod-textfield-icon-spacing-block-valid, var(--spectrum-textfield-icon-spacing-block-valid));
+    margin-block-end: var(--mod-textfield-icon-spacing-block-valid, var(--spectrum-textfield-icon-spacing-block-valid));
+    margin-inline-start: var(--mod-textfield-icon-spacing-inline-start-valid, var(--spectrum-textfield-icon-spacing-inline-start-valid));
+    margin-inline-end: var(--mod-textfield-icon-spacing-inline-end-valid, var(--spectrum-textfield-icon-spacing-inline-end-valid));
+    color: var(--highcontrast-textfield-icon-color-valid, var(--mod-textfield-icon-color-valid, var(--spectrum-textfield-icon-color-valid)));
+  }
+
+  /****** Validation Icon - Invalid ⚠️ ******/
+  .is-invalid & {
+    block-size: var(--mod-textfield-icon-size-invalid, var(--spectrum-textfield-icon-size-invalid));
+    inline-size: var(--mod-textfield-icon-size-invalid, var(--spectrum-textfield-icon-size-invalid));
+    margin-block-start: var(--mod-textfield-icon-spacing-block-invalid, var(--spectrum-textfield-icon-spacing-block-invalid));
+    margin-block-end: var(--mod-textfield-icon-spacing-block-invalid, var(--spectrum-textfield-icon-spacing-block-invalid));
+    margin-inline-start: var(--mod-textfield-icon-spacing-inline-start-invalid, var(--spectrum-textfield-icon-spacing-inline-start-invalid));
+    margin-inline-end: var(--mod-textfield-icon-spacing-inline-end-invalid, var(--spectrum-textfield-icon-spacing-inline-end-invalid));
+    color: var(--highcontrast-textfield-icon-color-invalid, var(--mod-textfield-icon-color-invalid, var(--spectrum-textfield-icon-color-invalid)));
+  }
+
+  .is-disabled &,
+  .is-readOnly & {
+    /* Disabled validation icons are transparent */
+    color: transparent;
+  }
+
+  .spectrum-Textfield--quiet & {
+    padding-inline-end: 0;
+  }
+
+  .spectrum-Textfield--quiet.is-valid & {
+    margin-inline-end: var(--mod-textfield-icon-spacing-inline-end-quiet-valid, var(--spectrum-textfield-icon-spacing-inline-end-quiet-valid));
+  }
+
+  .spectrum-Textfield--quiet.is-invalid & {
+    margin-inline-end: var(--mod-textfield-icon-spacing-inline-end-quiet-invalid, var(--spectrum-textfield-icon-spacing-inline-end-quiet-invalid));
+  }
+}
+
+/********* Child div - Input Wrapper - for alignment of focus indicator *********/
+.spectrum-Textfield-input-wrapper {
+  grid-row: 2;
+  grid-column: 1 / span 3;
+  box-sizing: border-box;
+  position: relative;
+
+  /*** Focus Indicator - Unfocused Base Styles ***/
+  &::after {
```

Comment:
In future we might be able to do away with pseudos for focus state indicators thanks to the support of `outline-offset` and `outline-style: auto` (safari rounded corners) and `outline-radius` (Firefox). 

```
outline: var(--mod-textfield-focus-indicator-width, var(--spectrum-textfield-focus-indicator-width)) auto var(--highcontrast-textfield-focus-indicator-color, var(--mod-textfield-focus-indicator-color, var(--spectrum-textfield-focus-indicator-color)));
outline-offset: (var(--mod-textfield-focus-indicator-gap, var(--spectrum-textfield-focus-indicator-gap);
```